internal/provider/resource/access_approval_policy.go

```diff
@@ -200,11 +202,12 @@ func (r *accessApprovalPolicyResource) Create(ctx context.Context, req resource.
 			Type: el.Type.ValueString(),
 		})
 	}
+	environments := infisicaltf.StringListToGoStringSlice(ctx, resp.Diagnostics, plan.EnvironmentSlugs)
 
 	accessApprovalPolicy, err := r.client.CreateAccessApprovalPolicy(infisical.CreateAccessApprovalPolicyRequest{
 		Name:              plan.Name.ValueString(),
 		ProjectSlug:       projectDetail.Slug,
-		Environment:       plan.EnvironmentSlug.ValueString(),
+		Environments:      environments,
```

Comment:
I think we should keep "Environment" there just in case the Infisical API server they are running on is not yet updated to the latest version

Reply:
review the other parts as well so that it works even when the API they've connected to is outdated